src/Utility/Function.h

```diff
@@ -251,6 +251,21 @@ class VecFunc : public utl::SpatialFunction<Vec3>, public FunctionBase
   {
     return this->evaluate(X).length();
   }
+
+  //! \brief Evaluates first derivatives of the function.
+  Tensor gradient(const Vec3& X) const
+  {
+    Tensor result(ncmp);
+    result = this->evalGradient(X);
+    return result;
+  }
+
+protected:
+  //! \brief Returns the gradient of the function as a 1D array.
+  virtual std::vector<Real> evalGradient(const Vec3& X) const
+  {
+    return {};
+  }
```

Comment:
The argument `X` is not used in this dummy implementation, and I prefer one-liner implementation:
`virtual std::vector<Real> evalGradient(const Vec3&) const { return {}; }`

Reply:
Another point, couldn't this implementation be placed in the parent class `SpatialFunction` instead? So you don't have to repeat it for each sub-class. Or are there some template issues that makes that difficult.

Reply:
Ok and yeah, I initial was working towards removing the SpatialFunction layer but ended up not doing it, so I have moved the slot into that thus dropping the duplicates of the dummy.